recipes/fcl/all/conanfile.py

```diff
@@ -47,10 +47,13 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("eigen/3.4.0")
-        self.requires("libccd/2.1")
+        # Used in fcl/common/types.h public header
+        self.requires("eigen/3.4.0", transitive_headers=True, transitive_libs=True)
+        # Used in fcl/narrowphase/detail/convexity_based_algorithm/support.h
+        self.requires("libccd/2.1", transitive_headers=True, transitive_libs=True)
         if self.options.with_octomap:
-            self.requires("octomap/1.9.7")
+            # Used in fcl/geometry/octree/octree.h
+            self.requires("octomap/1.9.7", transitive_headers=True, transitive_libs=True)
```

Comment:
none of these seem to require `transitive_libs` - eigen is a hader only library and for the other two headers mention, only types /macros seem to be required, but should cause no symbols to be _directly_ required by the translation unit including this header - at least, not "indirectly".

Reply:
It's generally not worth the time and effort to dig into the finer points of transitive_libs=True/False for every minor library, IMO, although I don't mind setting it to False for clear header-only libs like Eigen or octree.
It's very easy to miss uses of externally-defined symbols in the code and test_package usually tests a too narrow slice of the library to catch issues caused by transitive_libs=False.

Regarding Eigen specifically, though, it does have support for an optional SuiteSparse backend that could be well added in the future as an optional dependency, which is why I decided to keep it as True.

Anyway, I'll change it to False for Eigen and octree.

Reply:
> It's generally not worth the time and effort to dig into the finer points of transitive_libs=True/False for every minor library,

Thanks! We don't expect you to do it - we are very happy to do so as part of the review process so that we can orient this better to suit the recipe.

Reply:
Thanks!

> We can probably lint/statically detect the correct cases here - I'll make sure we have hooks that can advise better.

That would be absolutely incredible, but I imagine it's probably next to impossible to handle all the possible #ifdef configurations in the headers. Still, just knowing that transitive headers and possibly libs are definitely required with the baseline config would be very useful already.